ffs/api/api_deals.go

```diff
@@ -0,0 +1,74 @@
+package api
+
+import (
+	"fmt"
+
+	"github.com/textileio/powergate/deals"
+)
+
+// ListStorageDealRecords lists storage deals for this FFS instance according to the provided options.
+func (i *API) ListStorageDealRecords(opts ...deals.ListDealRecordsOption) ([]deals.StorageDealRecord, error) {
+	c := deals.ListDealRecordsConfig{}
+	for _, opt := range opts {
+		opt(&c)
+	}
+	finalAddrs, err := i.finalAddresses(c.FromAddrs)
```

Comment:
You can still provide a list of addresses that you want to query for, but the logic in `finalAddressees` will check that the addresses you specify are managed by the ffs instance. It returns an error if it finds an address not managed by the ffs instance